node/src/main/scala/coop/rchain/node/api/DeployGrpcService.scala

```diff
@@ -79,9 +77,19 @@ private[api] object DeployGrpcService {
         )
       }
 
+      override def machineVerifiableDag(q: MachineVerifyQuery): Task[GrpcEither] =
+        defer(BlockAPI.machineVerifiableDag[F])
+
       override def showBlocks(request: BlocksQuery): Observable[GrpcEither] =
         Observable
-          .fromTask(deferList(BlockAPI.showBlocks[F](request.depth)))
+          .fromTask(
+            deferList[BlockInfoWithoutTuplespace](
+              Functor[F].map(BlockAPI.showBlocks[F](Some(request.depth))) {
+                case Right(list) => list
+                case Left(_)     => List.empty
```

Comment:
Bifunctor's leftMap(_ => List.empty) is less verbose and less error prone than the explicit match